presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -566,6 +608,11 @@ public void testArrayAsJson()
                 identity());
     }
 
+    private static DataType<Entry<String, String>> hstoreArrayAsJsonDataType()
```

Comment:
This should be `DataType<Map<String, String>>` and below eg first lambda (currently `getKey`) should render SQL literal.
I know why you're doing this (it's simpler), but this breaks `DataType` contract and makes this ... surprising.

if you don't want to play by `DataType` rules, you can implement this test "by hand". 
eg https://github.com/prestosql/presto/commit/1f70a1b2af63a7c21354924b7464d13900315d55#diff-5de8e17fb3637b3f60e6d8f6972f64b3R294

Reply:
Thanks, that makes sense. I wanted to avoid parsing the JSON to construct the SQL literal, but I see now that it's required since we're doing a round trip. AFAICT, we can't use a map here since the output is a JSON string (the other hstore tests work because the SQL return type is a map).